### PR TITLE
Load Test

### LINKED ISSUE #1127: Load Test
### Problem

We need a Load Test Suite to be able to answer the following questions:

1. What is our current limit of "normal usage" users we can attend concurrently.
2. What are the ideal parameters to set an auto-scale policy in CPU and Memory.

### Solution

Extend the current k6 performance tests in place so we can use them as a first `Load Test`.

1. Load Test must call a sub-set of methods that represent the current usage of a single user.
2. Load Test must call all the methods concurrently as opposed to sequentially.
3. Load Test must should be run with different amount of VU's (10, 25, 50, 100, 150, 200, 500... until been able to answer Question 1)

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by AlfredoG87**
**Description**:
Improvements to k6 tests to be able to used them as Load Test

**Related issue(s)**:

Fixes #1127

**Notes for reviewer**:
While during the tests I was able to replicate the issue of `502` .

Mirror Node Latency:
image

Request Distribution:
image

Request Rate (RPS):
image

Relay Methods Latency:
image

# K6 Performance Test Results 

Timestamp: 2023-04-27T21:32:46.862Z 

Duration: 200s 

Test Type: load 

Virtual Users (VUs): 100 

| Scenario | VUS | Reqs | Pass % | RPS (1/s) | Pass RPS (1/s) | Avg. Req Duration (ms) | Median (ms) | Min (ms) | Max (ms) | P(90) (ms) | P(95) (ms) | Comment |
|----------|-----|------|--------|-----|----------|-------------------|-------|-----|-----|-------|-------|---------|
| eth_blockNumber | 100 | 817 | 61.64 | 4.00 | 2.47 | 4573.74 | 533.50 | 91.18 | 56817.96 | 13894.70 | 22725.88 | |
| eth_call | 100 | 780 | 58.54 | 3.81 | 2.23 | 5888.71 | 956.58 | 144.03 | 50591.10 | 20195.77 | 25528.24 | |
| eth_chainId | 100 | 903 | 65.63 | 4.41 | 2.89 | 2815.38 | 271.93 | 90.15 | 49949.48 | 7615.91 | 13264.06 | |
| eth_estimateGas | 100 | 811 | 61.65 | 3.96 | 2.44 | 3817.33 | 392.26 | 93.77 | 51322.57 | 11182.73 | 18761.71 | |
| eth_feeHistory | 100 | 778 | 60.93 | 3.81 | 2.32 | 4898.08 | 585.24 | 95.51 | 50511.11 | 15309.18 | 25038.32 | |
| eth_gasPrice | 100 | 855 | 64.48 | 4.18 | 2.70 | 3231.24 | 269.81 | 93.71 | 49431.81 | 9722.84 | 17122.53 | |
| eth_getBalance | 100 | 735 | 57.38 | 3.59 | 2.06 | 5066.47 | 584.07 | 95.46 | 54371.46 | 17570.78 | 28021.52 | |
| eth_getBlockByHash | 100 | 656 | 52.45 | 3.20 | 1.68 | 6662.04 | 922.73 | 193.02 | 52050.32 | 24445.93 | 29095.54 | |
| eth_getBlockByNumber | 100 | 770 | 59.66 | 3.76 | 2.24 | 5019.24 | 707.64 | 152.21 | 55419.14 | 19260.38 | 28013.14 | |
| eth_getCode | 100 | 748 | 60.11 | 3.65 | 2.19 | 4053.45 | 249.03 | 91.07 | 53301.43 | 13176.14 | 26475.02 | |
| eth_getTransactionCount | 100 | 724 | 54.28 | 3.53 | 1.92 | 5385.59 | 1485.32 | 251.69 | 50339.06 | 16581.35 | 25078.37 | |
| eth_getTransactionReceipt | 100 | 757 | 60.50 | 3.70 | 2.24 | 3393.69 | 260.04 | 92.19 | 57517.43 | 9867.86 | 16669.58 | |

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: k6/src/lib/common.js [hunk 1]
```diff
@@ -264,7 +282,16 @@ function TestScenarioBuilder() {
       run: function (testParameters, iteration = 0) {
         const response = that._request(testParameters, iteration);
         check(response, that._checks);
-      },
+
+        // if Load test, then we need to sleep for random time between 1 and 5 seconds
```
Comment (Nana-EC) [inline k6/src/lib/common.js:286]: Q: why?
Also can remove the random extra new lines

File: k6/src/scenarios/test/eth_call.js [hunk 2]
```diff
@@ -35,6 +35,8 @@ const {options, run} = new TestScenarioBuilder()
     )
   )
   .check(methodName, (r) => isNonErrorResponse(r))
+  .testDuration("3s")
```
Comment (Nana-EC) [inline k6/src/scenarios/test/eth_call.js:38]: Q: why were these needed?

**Review by Nana-EC — COMMENTED** [review]
LG, 2 Q's.

File: k6/src/scenarios/test/eth_call.js (hunk 2 above)
Comment (AlfredoG87) [reply k6/src/scenarios/test/eth_call.js:38]: Since `eth_call` is resolved using `mirror_node` response time can vary greatly and in order for the tests to pass, I custom override the default time to respond wich is 1 second (or defined by env property `DEFAULT_MAX_DURATION` to 3 seconds.

**Review by Nana-EC — DISMISSED** [review]
LG

File: k6/src/lib/common.js (hunk 1 above)
Comment (AlfredoG87) [reply k6/src/lib/common.js:285]: ✅  Done

**Review by Nana-EC — APPROVED** [review]
